x-pack/plugins/index_management/public/application/sections/enrich_policy_create/steps/configuration.tsx

```diff
@@ -0,0 +1,270 @@
+import { EuiButton, EuiText, EuiLink, EuiSpacer } from '@elastic/eui';
+import {
+  useForm,
+  Form,
+  fieldValidators,
+  FormSchema,
+  FIELD_TYPES,
+  UseField,
+  TextField,
+  SelectField,
+  JsonEditorField,
+} from '../../../../shared_imports';
+
+import { useAppContext } from '../../../app_context';
+import { IndicesSelector } from './fields/indices_selector';
+import { documentationService } from '../../../services/documentation';
+import { useCreatePolicyContext, DraftPolicy } from '../create_policy_context';
+
+interface Props {
+  onNext: () => void;
+}
+
+export const configurationFormSchema: FormSchema = {
+  name: {
+    type: FIELD_TYPES.TEXT,
+    label: i18n.translate('xpack.idxMgmt.enrichPolicyCreate.configurationStep.policyNameField', {
+      defaultMessage: 'Policy name',
+    }),
+    validations: [
+      {
+        validator: fieldValidators.emptyField(
+          i18n.translate(
+            'xpack.idxMgmt.enrichPolicyCreate.configurationStep.policyNameRequiredError',
+            {
+              defaultMessage: 'A policy name value is required.',
+            }
+          )
+        ),
+      },
+    ],
+  },
+
+  type: {
+    type: FIELD_TYPES.SELECT,
+    label: i18n.translate('xpack.idxMgmt.enrichPolicyCreate.configurationStep.policyTypeLabel', {
+      defaultMessage: 'Policy type',
+    }),
+    validations: [
+      {
+        validator: fieldValidators.emptyField(
+          i18n.translate('xpack.idxMgmt.enrichPolicyCreate.configurationStep.typeRequiredError', {
+            defaultMessage: 'A policy type value is required.',
+          })
+        ),
+      },
+    ],
+  },
+
+  sourceIndices: {
+    label: i18n.translate('xpack.idxMgmt.enrichPolicyCreate.configurationStep.sourceIndicesLabel', {
+      defaultMessage: 'Source indices',
+    }),
+    validations: [
+      {
+        validator: fieldValidators.emptyField(
+          i18n.translate(
+            'xpack.idxMgmt.enrichPolicyCreate.configurationStep.sourceIndicesRequiredError',
+            {
+              defaultMessage: 'At least one source index is required.',
+            }
+          )
+        ),
+      },
+    ],
+  },
+
+  query: {
+    type: FIELD_TYPES.JSON,
+    label: i18n.translate('xpack.idxMgmt.enrichPolicyCreate.configurationStep.queryLabel', {
+      defaultMessage: 'Query (optional)',
+    }),
+    serializer: (jsonString: string) => {
+      let parsedJSON: any;
+      try {
+        parsedJSON = JSON.parse(jsonString);
+      } catch {
+        parsedJSON = {};
+      }
+
+      return parsedJSON;
+    },
+    deserializer: (json: any) =>
+      json && typeof json === 'object' ? JSON.stringify(json, null, 2) : '{\n\n}',
+    helpText: (
+      <FormattedMessage
+        id="xpack.idxMgmt.enrichPolicyCreate.configurationStep.queryHelpText"
+        defaultMessage="Defaults to: {code} query."
+        values={{
+          code: (
+            <EuiLink external target="_blank" href={documentationService.getMatchAllQueryLink()}>
```

Comment:
I noticed that the "match_all" link leads to the current page (and not to the correct doc page) and it seems this is because `getMatchAllQueryLink()`, when called from here, returns an empty string. Probably `documentationService` should not be imported but passed as a prop?

Reply:
Nice catch! The documentationService isnt initialized at this point, so I moved it to the render method 🚀